src/ncp/rcp_host.cpp

```diff
@@ -424,10 +425,43 @@ void RcpHost::Leave(const AsyncResultReceiver &aReceiver)
 void RcpHost::ScheduleMigration(const otOperationalDatasetTlvs &aPendingOpDatasetTlvs,
                                 const AsyncResultReceiver       aReceiver)
 {
-    OT_UNUSED_VARIABLE(aPendingOpDatasetTlvs);
+    otError              error = OT_ERROR_NONE;
+    std::string          errorMsg;
+    otOperationalDataset emptyDataset;
 
-    // TODO: Implement ScheduleMigration under RCP mode.
-    mTaskRunner.Post([aReceiver](void) { aReceiver(OT_ERROR_NOT_IMPLEMENTED, "Not implemented!"); });
+    VerifyOrExit(mInstance != nullptr, error = OT_ERROR_INVALID_STATE, errorMsg = "OT is not initialized");
+    VerifyOrExit(IsAttached(), error = OT_ERROR_FAILED,
+                 errorMsg = "Cannot schedule migration when this device is detached");
+
+    // TODO: check supported channel mask
+
+    SuccessOrExit(error    = otDatasetSendMgmtPendingSet(mInstance, &emptyDataset, aPendingOpDatasetTlvs.mTlvs,
+                                                         static_cast<uint8_t>(aPendingOpDatasetTlvs.mLength),
+                                                         SendMgmtPendingSetCallback, this),
+                  errorMsg = "Failed to send MGMT_PENDING_SET.req");
+
+exit:
+    if (error != OT_ERROR_NONE)
+    {
+        mTaskRunner.Post([aReceiver, error, errorMsg](void) { aReceiver(error, errorMsg); });
+    }
+    else
+    {
+        // otDatasetSendMgmtPendingSet() returns OT_ERROR_BUSY if it has already been called before but the
+        // callback hasn't been invoked. So we can guarantee that mMigrationReceiver is always nullptr here
+        assert(mScheduleMigrationReceiver == nullptr);
```

Comment:
Should we use `OT_ASSERT` here?

Reply:
I think we should just use `assert` because `OT_ASSERT` is within OT core (`openthread/src/core/common/debug.hpp`). We try to avoid including OT core headers in ot-br-posix because that may cause conflicts. In ot-br-posix `assert` is already used in many places.